cloud/aws/modules/pgadmin/task.tf

```diff
@@ -0,0 +1,151 @@
+resource "aws_ecs_task_definition" "pgadmin" {
+  family = "${local.name_prefix}-pgadmin"
+
+  cpu    = 1024
+  memory = 2048
+
+  container_definitions = jsonencode([
+    module.pgadmin_init_container_def.json_map_object,
+    module.pgadmin_container_def.json_map_object
+  ])
+
+  execution_role_arn       = aws_iam_role.civiform_pgadmin_task_execution_role.arn
+  network_mode             = "awsvpc"
+  requires_compatibilities = ["FARGATE"]
+
+  volume {
+    name = "config"
+  }
+
+  tags = local.tags
+}
+
+module "pgadmin_container_def" {
+  source  = "cloudposse/ecs-container-definition/aws"
+  version = "0.58.1"
+
+  container_name  = "${var.app_prefix}-pgadmin"
+  container_image = "dpage/pgadmin4:6.15"
+  container_depends_on = [
+    { containerName = "${var.app_prefix}-init", condition = "COMPLETE" }
+  ]
+
+  secrets = [
+    {
+      name      = "PGADMIN_DEFAULT_EMAIL"
+      valueFrom = aws_secretsmanager_secret_version.pgadmin_username_secret_version.arn
+    },
+    {
+      name      = "PGADMIN_DEFAULT_PASSWORD"
+      valueFrom = aws_secretsmanager_secret_version.pgadmin_password_secret_version.arn
+    }
+  ]
+
+  map_environment = {
+    PGADMIN_SERVER_JSON_FILE = "/config/servers.json"
+  }
+
+  port_mappings = [
+    {
+      containerPort = 80
+      hostPort      = 80
+      protocol      = "tcp"
+    },
+  ]
+
+  mount_points = [
+    {
+      containerPath = "/config",
+      sourceVolume  = "config",
+      readOnly      = false
+    }
+  ]
+
+  log_configuration = {
+    logDriver = "awslogs"
+    options = {
+      "awslogs-region"        = var.aws_region
+      "awslogs-stream-prefix" = "ecs"
+      "awslogs-group"         = module.pgadmin_logs.logs_path
+      "awslogs-create-group"  = "true"
+      # Use https://docs.docker.com/config/containers/logging/awslogs/#awslogs-multiline-pattern
+      # Logs are streamed via container's stdout. Each line is considered a
+      # separate log messsage. To collect stacktraces, which take multiple line,
+      # to a single event we consider all lines which start with a whitespace character to be
+      # part of the previous line and not a standalone event.
+      "awslogs-multiline-pattern" = "^[^\\s]"
+    }
+    secretOptions = null
+  }
+}
+module "pgadmin_logs" {
+  source    = "cn-terraform/cloudwatch-logs/aws"
+  version   = "1.0.12"
+  logs_path = "${var.app_prefix}-pgadmin-logs/"
+  tags = {
+    Name = "${var.app_prefix} pgAdmin Cloud Watch Logs"
+    Type = "pgAdmin Cloud Watch Logs"
+  }
+}
+
+module "pgadmin_init_container_def" {
```

Comment:
Can you add comment explaining the purpose of this container? Given that it's pretty odd container (just used to generate json file).

Reply:
Done.

Reply:
Another question related to that. Is there a race condition between the first container (that writes server.json) and the second (pgadmin)? Meaning that first container must run before the second initializes?

Reply:
There would be a race condition if I did not configure:

```
  container_depends_on = [
    { containerName = "${local.name_prefix}-init", condition = "COMPLETE" }
  ]
```

in the pgadmin_container_def

Reply:
Ah, cool!